### PR TITLE
Content refactoring start - checkStatus POC

### PR DESCRIPTION
**Pull request by beshur**
Refactoring based on functions:

- [x] Move `checkStatus` into class
- [ ] Move `start` into class
- [ ] Move `pause` into class
- [ ] Move `getTitle` into class
- [ ] index.js cleanup

### CODE REVIEW

File: src/content/index.js [hunk 1]
```diff
@@ -130,13 +133,6 @@ class Provider {
         this._detectProviderAndStartCheckInterval();
     }
 
-    isInstalled() {
-        if (window.location.host.replace('www.', '') == 'stoplay_page.dev'
-            || window.location.host.replace('www.', '') == 'stoplay.github.io') {
-            document.querySelector("body").className = document.querySelector("body").className + " m_installed";
-        }
-    }
-
```
Comment (beshur) [inline src/content/index.js]: this is obsolete

File: src/content/ProviderCheckStatus.js [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+
+  setHost(host) {
+    this.host = host;
+  }
+
+  check() {
+    console.log('providercheckStatus', this);
+    let status, p, selector, selectorQuery;
+
+    switch(this.host) {
```
Comment (beshur) [inline src/content/ProviderCheckStatus.js]: Moved out of index.js without changes

File: src/content/mixins/oneOfTheVideos.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+/* A shared mixin for when there is a video tag on page */
+import { Status } from '../Status.Types.js';
+
+export const checkStatus = () => {
```
Comment (endway) [inline src/content/mixins/oneOfTheVideos.js]: I like the idea to have mixins, as most of the status/pause/play functions can be grouped into few mixins, like:

**status:**
```js
document.querySelector(PROVIDER_SPECIFIC_QUERY).classList.contains(PROVIDER_SPECIFIC_CLASS)
```

**play/pause:**
```js
document.querySelector(PROVIDER_SPECIFIC_QUERY).click();
```

But, what I don't like is creating a separate class for each provider.
Previously, I thought that this is ok, but now it seems to me that that's a lot of redundant classes.

I'm thinking about a Strategy pattern, which may fix this:
```js
class Service {
    constructor(statusStrategy, controlStrategy, titleStrategy) {
    }
    
    getStatus() {
        return this.statusStrategy.getStatus();
    }

    play() {
        this.controlStrategy.play();
    }

    pause() {
        this.controlStrategy.pause();
    }
}

const servicesRegistry = [
    {
        hostPattern: "audible.**", // we can add support fields, like `host: string`, `hosts: Array<string>`
        statusStrategy: oneOfeTheVideos.getCheckStatusStrategy(".path .to .player"),
        controlStrategy: someOtherType.getControlStrategy("#player"),
    },
];

function getService(domain) {
    const matchedService = servicesRegistry.find(serviceConfig => {
        return serviceConfig.host === domain
            || serviceConfig.hosts.include(domain)
            || serviceConfig.hostPattern.match(domain);
    });

    if (!matchedService) {
        return;
    }

    return new Service(matchedService.statusStrategy, matchedService.controlStrategy);
}
```

Not sure about registry.
@beshur What do you think?

File: src/content/ProviderCheckStatus.js [hunk 4]
```diff
@@ -0,0 +1,317 @@
+/* Checking status for all providers */
+import {Status} from './Status.Types.js';
+
+export class ProviderCheckStatus {
+  constructor() {
+    this.host = null;
+    this.customLastPlayerSelector = null;
```
[the remark this answers is not in the document]
Comment (beshur) [reply src/content/ProviderCheckStatus.js]: I wanted to pass it as a param to provider methods

**Comment by beshur** [comment]
Clean branch #157

**Pull request closed without merging**